src/Neo/SmartContract/Native/PolicyContract.cs

```diff
@@ -258,6 +269,98 @@ public bool IsBlocked(IReadOnlyStore snapshot, UInt160 account)
             return snapshot.Contains(CreateStorageKey(Prefix_BlockedAccount, account));
         }
 
+        internal bool IsWhitelistFeeContract(DataCache snapshot, UInt160 contractHash, string method, int argCount, [NotNullWhen(true)] out long? fixedFee)
+        {
+            // Check contract existence
+
+            var currentContract = ContractManagement.GetContract(snapshot, contractHash);
+
+            if (currentContract != null)
+            {
+                // Check state existence
+
+                var item = snapshot.TryGet(CreateStorageKey(Prefix_WhitelistedFeeContracts, contractHash, method, argCount))
+                    ?.GetInteroperable<WhitelistedFeeContract>();
+
+                if (item != null)
+                {
+                    // Check UpdateCounter
+
+                    if (item.UpdateCounter == currentContract.UpdateCounter)
+                    {
+                        fixedFee = item.FixedFee;
+                        return true;
+                    }
```

Comment:
Just a suggestion: we may automatically remove contract from the list of whitelisted on every contract update. This will require ContractManagement modification, but then committee won't be required to unregister whitelisted contract manually after its update.